packages/circuit-ui/components/ListItem/ListItem.tsx

```diff
@@ -358,10 +357,8 @@ export const ListItem = forwardRef(
 
     const isInteractive = !!props.href || !!props.onClick;
     const isNavigation = variant === 'navigation';
-    const hasSuffix = !!suffixLabel || !!suffix;
-    const hasInvalidSuffix = hasOnlySuffixDetails || hasCustomAndLabelSuffix;
-    const shouldRenderSuffixContainer =
-      !hasInvalidSuffix && (hasSuffix || isNavigation);
```

Comment:
I'd suggest removing this safeguard since we're already rendering warnings, and don't think we need to guard against bad use of the component to this extent.

Another issue with this is that the trailing element wouldn't be shown, and since the console warning can be missed depending on how much noise there is in it, a developer could be surprised by the behavior and take more time to fix her implementation than if the trailing element was rendered but looked wrong (e.g. in terms of alignment).